WebApplicationDemo/Controllers/FlurlDemoController.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+using Ark.Tools.Core;
+using Ark.Tools.Solid;
+
+using Asp.Versioning;
+
+using Flurl.Http;
+using Flurl.Http.Configuration;
+using Flurl.Http.Newtonsoft;
+
+using Microsoft.AspNetCore.Mvc;
+
+using WebApplicationDemo.Api.Queries;
+using WebApplicationDemo.Api.Requests;
+using WebApplicationDemo.Dto;
+
+using System.Text.Json;
+using System.Collections.Generic;
+
+namespace WebApplicationDemo.Controllers
+{
+    [Route("flurl-demo")]
+    [ApiVersion(3.0)]
+    public class FlurlDemoController : ApiController
+    {
+        private readonly IFlurlClientCache _flurl;
+        private string _url;
+
+        public FlurlDemoController(IFlurlClientCache flurl) 
+        {
+            _flurl = flurl;
+            _url = "https://jsonplaceholder.typicode.com/";
+        }
+
+        [Route("posts")]
+        [HttpGet]
+        public async Task<IActionResult> Get()
+        {
+            var client = _flurl.GetOrAdd(_url, _url);
```

Comment:
client is IDisposable thus shoukd be Disposed.

I'm quite sure that doesn't work well with a 'cache' pattern.

Also, the usage patter should enable the creation and customization of the Client by each external service.

Please rework implementing the full pattern with:

1. A request handler 
2. An external service singleton where flurl is used 
3. dispose disposables 
4. customize the client once, e.g. in the ctor.
5. use the FullName of the Service class as key, not the url, to demo a less error-prone approach (url is not bad but I know already of cases where would not work). The idea is that the Client instance is "owned" by the Service instance.

Reply:
According to this [docs ](https://flurl.dev/docs/clients/)I think the cache is responsible for disposing the clients

Reply:
add service